src/bam/mod.rs

```diff
@@ -811,8 +822,8 @@ impl HeaderView {
     }
 
     #[inline]
-    pub fn inner(&self) -> htslib::bam_hdr_t {
-        unsafe { (*self.inner) }
+    pub fn inner(&self) -> &htslib::bam_hdr_t {
+        unsafe { self.inner.as_ref().unwrap() }
     }
```

Comment:
Like, this was weird. `inner()` wasn't returning a reference to the interior, but a copy of the object.

Some places would access a member of the copy, still others would pass that copy as a mutable reference into htslib functions.